### ISSUE TITLE
abnf.js", which threw an exception: TypeError: Cannot read property 'decode' of undefined

### ISSUE THREAD

**Issue by Jonathansoufer** [spruceid/siwe#165]
Trying to use ParsedMessage from '@spruceid/siwe-parser'; on our React Native app to parse the payload string received from WalletConnect. The error can be reproduced by implementing the snippet below and trying `https://login.xyz/`.

```
export function parseSiweMessageRequest(message: string) {
  try {
    const parsedMessage = new ParsedMessage(message);

    return parsedMessage;
  } catch (error) {
    console.log({ parsedMessage: error });
    return null;
  }
}
```

![Screenshot 2023-04-24 at 3 02 15 PM](https://user-images.githubusercontent.com/44679989/234019916-efc276ab-0d8c-411a-9487-9e18ec6f8fbf.png)

**Comment by w4ll3** [comment]
Hello, glad to have your feedback. `siwe` libraries require some node polyfills, I'm afraid you might be missing them. You might want to use `[SSX](https://docs.ssx.id)` which already includes those and also give you a lot more on top. In the same repo there are a lot of React examples not sure if they can help you out.

**Comment by Jonathansoufer** [comment]
Thank you for your quick reply. I am afraid that doesn't solve the problem. In the scenario, I'm the wallet not the dApp itself. So ssx is used by the dApp to establish with the respective backend all the SIWE flow, but as a wallet using the WalletConnect lib (https://walletconnect.com/) to connect into dApps. From that perspective I just need to parse a request and be able to identify if it's a SIWE or not and produce an object out of the string payload received. I can do it via custom implementation but seems that @spruceid/siwe-parser implements that. What am I missing to make this work with React NATIVE?

**Comment by Jonathansoufer** [comment]
Team work!

**Comment by luneShaoGM** [comment]
Same with me

**Comment by ldthomas** [comment]
I've just published `apg-js` version 4.2.0 specifically modified to address the fix that @Jonathansoufer pointed out and more. Give 4.2.0 a try and if it fixes this problem then I would assume this issue could be closed.
